Disable refund button when transaction disputed

shendy-a8c opened the pull request:
Fixes https://github.com/Automattic/woocommerce-payments/issues/6378

#### Changes proposed in this Pull Request

- Disable refund button when there is active or lost dispute.
- Show explanation of to why refund button is disabled:

![Untitled](https://github.com/Automattic/woocommerce-payments/assets/73803630/9ce2cb20-fc69-4203-8beb-b9cd02ddc91b)

I refactored `DisputeNotice` a bit so it can show the explanation when refund button is disabled.
Follow the commits in this PR to understand the diff better.

When dispute is closed (status `warning_closed`), [per Stripe's docs](https://stripe.com/docs/disputes/how-disputes-work#:~:text=If%20an%20inquiry%20has%20been%20open%20for%20120%20days%20without%20escalation%20to%20a%20chargeback%2C%20Stripe%20marks%20it%20as%20closed%20in%20the%20Dashboard%20and%20API.%20At%20this%20point%2C%20you%20can%20be%20confident%20the%20card%20network%20won%E2%80%99t%20escalate%20it%E2%80%94they%20don%E2%80%99t%20provide%20an%20explicit%20%E2%80%9Cwin%E2%80%9D%20message%20for%20inquiries.), looks like refund should be allowed although I cannot test this for real. I do not disable the refund button in such case. Worst case, if Stripe does not allow refund, user will see the error.

![Screenshot 2023-07-27 at 09 14 41](https://github.com/Automattic/woocommerce-payments/assets/73803630/45cc330d-e287-4b33-9d6d-f6ca0a28e404)

#### Testing instructions

Either create transaction with different state of disputes or alter the dispute object [here](https://github.com/Automattic/woocommerce-payments/blob/5be7cb9e3485cf59915a3a562dc9305e51c5a63a/client/order/index.js#L138). I tested by altering the dispute object [here](https://github.com/Automattic/woocommerce-payments/blob/5be7cb9e3485cf59915a3a562dc9305e51c5a63a/client/order/index.js#L138), eg:
```
dispute.status = 'won';
```

- When there is no dispute:
Dispute notice should not show. Refund button should not be disabled.

- When there is a dispute needing a response:
Warning text and CTA button should show. Refund button should be disabled and there should be the explanation of to why the refund button is disabled.

![Screenshot 2023-08-23 at 01 57 23](https://github.com/Automattic/woocommerce-payments/assets/73803630/6fa11fea-57dc-41e1-875f-c14b83211cce)

- When there is a dispute that does not need a response, eg when its status is `under_review`, or when due date has passed:
Warning text and CTA button should not show. Refund button should be disabled and there should be the explanation of to why the refund button is disabled.

![Screenshot 2023-08-23 at 02 00 46](https://github.com/Automattic/woocommerce-payments/assets/73803630/baf682aa-bfd7-4beb-8ef0-ea8b071aeb62)

- When there is a dispute won / closed (status `warning_closed`):
Warning text and CTA button should not show. Refund button should not be disabled.
`ps:` you can submit a winning evidence by putting `winning_evidence` under the 'Additional details' field ([ref](https://stripe.com/docs/testing#evidence)) of the dispute challenge form.

![Screenshot 2023-08-23 at 02 22 17](https://github.com/Automattic/woocommerce-payments/assets/73803630/a93f2012-fe05-4d12-badc-8ab091419014)


-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

<!--
Make sure you edit the page for the current release when adding testing instructions.
We often create a blank page ahead of time for the next release.
If this PR need not be QA tested, edit to 'QA Testing Not Applicable'
-->

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

shendy-a8c committed:
Return fast if there is no dispute.

shendy-a8c committed:
Move dispute variable assignment.

shendy-a8c committed:
Move urgency variable to the top.

shendy-a8c committed:
Use actions variables.

shendy-a8c committed:
Disable refund button and show explanation.

shendy-a8c committed:
showWarning flag.

shendy-a8c committed:
Use warningText variable.

shendy-a8c committed:
Put code around warning text inside the if clause.

shendy-a8c committed:
No need optional chaining charge because at that point, dispute must have been defined.

shendy-a8c committed:
Return null if no warning text and refund is not disabled.

shendy-a8c committed:
Changelog entry.

botwoo commented:
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7043` or branch name `update/6378-disable-refund-button-when-disputed` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=update/6378-disable-refund-button-when-disputed) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 687592ba445cace3b2ce2c5242135a3adbe15fc4
- Build time: 2023-09-05 11:01:08 UTC

_Note: the build is updated when a new commit is pushed to this PR._

shendy-a8c commented:
@nikkivias I take a stab on how to solve this problem. Please read the Testing Instructions section in the PR body.

brucealdridge reviewed (commented):
Refunding is allowed if the dispute is an inquiry. This is a [recommended way of resolving an inquiry](https://stripe.com/docs/disputes/how-disputes-work#:~:text=or%20by%20issuing%20a%20full%20refund).

brucealdridge commented on client/order/index.js:
```diff
@@ -243,11 +256,32 @@ const DisputeNotice = ( { chargeId } ) => {
+			{ disableRefund && (
+				<div>
+					{ __(
+						'Refunding and order editing are disabled during disputes to maintain record integrity.',
```
Putting my merchant hat on, 'maintain record integrity' wouldn't mean anything to me.

```suggestion
						'Refund and order editing options are unavailable at the moment because this order is in dispute.',
```

brucealdridge commented:
I think we should update this tooltip as well as it sends a conflicting message

![image](https://github.com/Automattic/woocommerce-payments/assets/57298/b0512cf0-e7ec-4bb2-89f8-32f3c46c4496)

nikivias commented:
@brucealdridge acknowledging your comment, I removed the last phrase to keep the message brief.

thank you @shendy-a8c for getting these started. I mocked them with some word changes. cc: @Jinksi 

Needs response:

> This order has a payment dispute of $120.00 USD labeled as 'Purchase unauthorized'. Please respond before May 29, 2023. Refunds and order editing are disabled during disputes.

Under review:

> This order has a payment dispute. Refunds and order editing are disabled during disputes. View dispute

<img width="1440" alt="Container" src="https://github.com/Automattic/woocommerce-payments/assets/99638528/9ffa10f8-6c94-452f-bc6f-205284979e5f">
<img width="1440" alt="Container-1" src="https://github.com/Automattic/woocommerce-payments/assets/99638528/dda43353-ff14-4d6d-8875-1f8653b8a83d">

[I posted a question on Slack](https://a8c.slack.com/archives/C03LA109BKM/p1692768707304979) on how we should address post dispute stages.

Agree with @brucealdridge that the tooltip for _This order is no longer editable_ is conflicting. So I suggest we use the same line:

> Refunds and order editing are disabled during disputes.

Curious if they can edit orders post dispute though? That line _no longer editable_ is throwing me off

<img width="547" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/99638528/57b6d557-0af5-4169-a866-4d764b4e9008">

shendy-a8c committed:
Merge branch 'develop' into update/6378-disable-refund-button-when-disputed

shendy-a8c committed:
Merge branch 'develop' into update/6378-disable-refund-button-when-disputed

shendy-a8c committed:
Refund is allowed for order that is disputed with status `warning_needs_response`, `warning_under_review`, or `warning_closed`, ie dispute with type inquiry.

shendy-a8c committed:
Show different notice text about disabled refunds when dispute needs a response or not.

shendy-a8c committed:
Merge branch 'develop' into update/6378-disable-refund-button-when-disputed

shendy-a8c committed:
Create a helper function called `isUnderReview()` to determine whether a dispute status is under review.

shendy-a8c committed:
Make sure refund button can be found and change the refund tooltip.

shendy-a8c committed:
When dispute is closed, it's not refundable.

shendy-a8c commented:
@nikkivias 'Refunds and order editing are disabled during disputes.' sounds good when there is active dispute but what about when dispute is lost?

The word 'during' does not sound suitable when the dispute has lost. For now, I use:
> Refunds and order editing are disabled because this order was disputed. View dispute

Let me know if that needs to change to something else.

shendy-a8c commented on client/order/index.js:
```diff
@@ -130,102 +131,177 @@ jQuery( function ( $ ) {
+			refundButton.disabled = true;
+
+			const disputeDetailsLink = getDetailsURL( dispute.id, 'disputes' );
+			// TODO: finalize text copy after we come up with the final decision.
```
This TODO needs to be resolved.

See https://github.com/Automattic/woocommerce-payments/pull/7043#issuecomment-1694720752

Jinksi replied:
Has this been resolved now?

shendy-a8c commented on client/order/index.js:
```diff
@@ -130,102 +131,177 @@ jQuery( function ( $ ) {
-		reasons[ dispute.reason ].display,
-		dateI18n( 'M j, Y', dueBy.local().toISOString() )
-	);
+			// TODO: use isAwaitingResponse() from https://github.com/Automattic/woocommerce-payments/pull/6998.
```
Waiting on https://github.com/Automattic/woocommerce-payments/pull/6998.

shendy-a8c commented:
> Curious if they can edit orders post dispute though?

Yes, order can still be edited. For example, you can change the order's status.

<img width="418" alt="Screenshot 2023-08-28 at 01 31 01" src="https://github.com/Automattic/woocommerce-payments/assets/73803630/b099e306-67f7-4941-af8f-59030c1844ce">

In fact, the suggestion showing in the tooltip as pointed out by @brucealdridge's comment https://github.com/Automattic/woocommerce-payments/pull/7043#issuecomment-1689269216 is to change the order's status to 'Pending Payment'.

shendy-a8c commented:
I observed these warnings:

<img width="1521" alt="Screenshot 2023-08-28 at 01 00 26" src="https://github.com/Automattic/woocommerce-payments/assets/73803630/9c48f486-f03a-4e12-badd-b7f4ed4f3012">

So I created https://github.com/Automattic/woocommerce-payments/issues/7074.

shendy-a8c commented:
While waiting on some decisions, this PR is ready for another round of review.

shendy-a8c committed:
Merge branch 'develop' into update/6378-disable-refund-button-when-disputed

# Conflicts:
#	client/disputes/utils.ts
#	client/order/index.js

Jinksi commented on client/order/index.js:
```diff
@@ -16,10 +16,15 @@ import BannerNotice from 'wcpay/components/banner-notice';
 import { useCharge } from 'wcpay/data';
 import wcpayTracks from 'tracks';
 import './style.scss';
+import { createInterpolateElement } from '@wordpress/element';
```
Nit: this import should be under "External dependencies"

shendy-a8c replied:
Addressed in 201674be5cd4e6b5f62d7490a3182ed7de0e1e00.

Jinksi commented on client/order/index.js:
```diff
@@ -129,102 +134,176 @@ jQuery( function ( $ ) {
+					continue;
+				}
+				jQuery( tooltip ).tipTip( { content: tooltipText } );
+			}
```
Minor suggestion: this might be more readable if we handle all of the logic with jQuery, since we're using jQuery for `tipTip` anyway – happy to stick with your vanilla method though.

I also notice that the `aria-label` attribute should also be updated to include the new tooltip content.

```suggestion
			jQuery( refundButton )
				.parent()
				.find( '.woocommerce-help-tip' )
				.attr( {
					// jQuery.tipTip uses the title attribute to generate the tooltip.
					title: tooltipText,
					'aria-label': tooltipText,
				} )
				// Regenerate the tipTip tooltip.
				.tipTip();
```

shendy-a8c replied:
Good idea. Addressed in https://github.com/Automattic/woocommerce-payments/pull/7043/commits/54b7199da469a483e4c7b22ed064704b379b3158.

Jinksi commented on client/order/index.js:
```diff
@@ -129,102 +134,176 @@ jQuery( function ( $ ) {
+		const countdownDays = Math.floor( dueBy.diff( now, 'days', true ) );
+
+		// If the dispute is due in the past, we don't want to show the notice.
+		if ( ! now.isAfter( dueBy ) ) {
```
Minor suggestion: use `isBefore` to improve readability.
```suggestion
		if ( now.isBefore( dueBy ) ) {
```

Jinksi commented on client/order/index.js:
```diff
@@ -129,102 +134,176 @@ jQuery( function ( $ ) {
-	const dueBy = moment.unix( dispute.evidence_details?.due_by );
-	const countdownDays = Math.floor( dueBy.diff( now, 'days', true ) );
+	let urgency = 'warning';
+	let actions = [];
```
Suggestion: initialise `actions` as `undefined` to let the `BannerNotice` know not to render the actions container element, removing unnecessary padding.

```suggestion
	let actions;
```

**Before**
![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/16452026-b573-485b-8c65-949687830a4c)


**After**
![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/d5c2cf9b-0d5a-45d8-aad4-6c2bc204c6b1)

shendy-a8c replied:
Nice!

Jinksi reviewed (changes requested):
LGTM @shendy-a8c 🏅 aside from the BannerNotice padding bug https://github.com/Automattic/woocommerce-payments/pull/7043#discussion_r1308082154.

I've left some other minor and optional suggestions.

This should be good to merge after that change and once we confirm text content.

### Dispute statuses I tested

**Needs Response**
Refunds disabled ✅
![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/567b807f-b6f1-4e8d-a91c-37c6426fbdb9)

**Under Review**
Refunds disabled ✅
![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/8a8d3077-20ef-4260-9151-db4d4e9ac096)

**Won**
No notice – refunds allowed.

**Lost**
Refunds disabled ✅
![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/77096706-6594-4985-b38a-caf80a39f020)

**Inquiry: Needs Response**
Refunds allowed ✅

![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/78b5e4a7-f6f6-4030-be78-9e3f6cf23155)

**Inquiry: Under Review**
No notice – refunds allowed.

**Inquiry: Closed**
Refunds disabled ❓ – check that this is correct. Since I understand this to mean that a refund has occurred **OR** the bank dropped the inquiry for another reason.
![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/cc776840-3dc3-456a-9e40-c65828fbf9dd)

shendy-a8c committed:
Pass undefined actions to `BannerNotice` so it does not render unnecessary padding.

Co-authored-by: Eric Jinks <3147296+Jinksi@users.noreply.github.com>

shendy-a8c committed:
Merge branch 'develop' into update/6378-disable-refund-button-when-disputed

shendy-a8c committed:
Allow refund if dispute is `warning_closed`.

Dispute becomes `warning_closed` due to several reasons: charge is fully refunded, or some other reasons.
In the case of the latter, we would still want to allow refunds.

shendy-a8c committed:
Move `createInterpolateElement` import from `@wordpress/element` under the external dependencies section.

Jinksi commented on client/disputes/utils.ts:
```diff
@@ -58,6 +61,10 @@ export const isAwaitingResponse = (
+	return disputeUnderReviewStatuses.includes( status );
+};
+
 export const isInquiry = ( dispute: Dispute | CachedDispute ): boolean => {
```
Not a part of this PR, but I see that `isInquiry` accepts a `Dispute` object. This is inconsistent with `isAwaitingResponse` and `isUnderReview` that accept `DisputeStatus`.

IMO we should change `isInquiry` to accept `DisputeStatus` to be consistent with the other util functions since the function doesn't need any other dispute details.

shendy-a8c replied:
I'll create a separate issue for this.

shendy-a8c replied:
https://github.com/Automattic/woocommerce-payments/issues/7085

shendy-a8c committed:
When updating refund tooltip, use jQuery to select the refund tooltip element to improve code readability.

shendy-a8c committed:
Use `isBefore()` instead of `! isAfter()` to improve readability.

Co-authored-by: Eric Jinks <3147296+Jinksi@users.noreply.github.com>

shendy-a8c commented:
@nikkivias Please confirm the copy I suggested here https://github.com/Automattic/woocommerce-payments/pull/7043#issuecomment-1694720752. I think that is the only pending thing in this PR.

shendy-a8c commented:
Thank you for the extensive testing and detailed review, @brucealdridge  @Jinksi !

Jinksi commented:
I've checked out this branch on the [Helix demo site](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/admin.php?page=wc-admin&path=/payments/overview) – here are links to disputed orders in various states (see PdjTHR-2T4-p2 for more links to example disputes):

* [Needs Response](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/post.php?post=137&action=edit)
	* [x] Refunds disabled
	* [x] Notice rendered
* [Under Review](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/post.php?post=108&action=edit)
	* [x] Refunds disabled
	* [x] Notice rendered
* [Won](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/post.php?post=76&action=edit)
	* [x] No notice
	* [x] Refunds allowed
* [Lost](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/post.php?post=106&action=edit) (losing evidence)
	* [x] Refunds disabled
	* [ ] Notice rendered ⚠️ **No notice or message rendered**
* [Lost](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/post.php?post=121&action=edit) (accepted)
	* [x] Refunds disabled
	* [ ] Notice rendered ⚠️ **No notice or message rendered**
* [Inquiry: Needs Response](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/post.php?post=138&action=edit)
	* [x] Refunds allowed
	* [x] Notice rendered
* [Inquiry: Under Review](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/post.php?post=139&action=edit)
	* [x] Refunds allowed
	* [x] No notice ❓ **Should we render a notice explaining this dispute status?**
* [Inquiry: Closed](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/post.php?post=74&action=edit) (refunded)
	* [x] Refunds allowed (although $0 left to refund)
	* [x] No notice
* [Inquiry: Closed](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/post.php?post=140&action=edit) (winning evidence)
	* [x] Refunds allowed (with full amount available to refund)
	* [x] No notice

@shendy-a8c should the `lost` disputes have a notice rendered, or is it obvious enough since the order status is **Refunded** already?

Jinksi commented:
Note: `BannerNotice` has been renamed `InlineNotice` in #7078.

shendy-a8c commented:
Thank you laying out all those scenarios on the demo site, @Jinksi.

> @shendy-a8c should the lost disputes have a notice rendered, or is it obvious enough since the order status is Refunded already?

From my understanding, no notice should be rendered because this PR focuses on more the explanation text for the disabled refund button, so only that copy text that will be rendered when dispute is lost.

Notice only shows up when dispute needs response. That behaviour remains the same even after this PR.

I think @nikkivias was also wondering if we should display the dispute notice when it is not needing a response per https://github.com/Automattic/woocommerce-payments/pull/7043#issuecomment-1689314389
> I posted a question on Slack on how we should address post dispute stages.

I don't see definite decision on that Slack thread, so I don't make any change around the dispute notice.

shendy-a8c commented:
@nikkivias there are 2 things we need your feedback:

1. Please confirm the disabled refund notice when dispute is lost.

Context: https://github.com/Automattic/woocommerce-payments/pull/7043#issuecomment-1694720752

```diff
- Refunds and order editing are disabled during disputes.
+ Refunds and order editing are disabled because this order was disputed.
```

2. Do we need to show dispute notice when dispute does not need a response?

Context: https://github.com/Automattic/woocommerce-payments/pull/7043#issuecomment-1698520828

Dispute notice is squared in red.
Disabled refund notice is in green.

<img width="1039" alt="Screenshot 2023-08-30 at 13 29 44" src="https://github.com/Automattic/woocommerce-payments/assets/73803630/896e8278-a038-4e3f-aa39-7614eb1e9ce5">

nikivias commented:
@shendy-a8c I appreciate the summary as I was getting a little woozy 😵‍💫 hehe

Based on our Slack conversation, I believe there's consensus that orders should have a clear indicator when a refund is not or no longer an option. So yes we should display a notice, to sort of 'mark' an order, but the CTA should just be a link and not a button. 

I've adjusted the texts depending on status accordingly. I changed the CTA to 'View details' because it leads to the Transaction page where the dispute is only a part of it. 

`Under review`
> This order has an active payment dispute. Refunds and order editing are disabled during this time. [View details]

Tooltip: Refunds and order editing are disabled during an active dispute. 

`Accepted`, and `Lost`
> Refunds and order editing have been disabled as a result of a lost dispute. [View details]

Tooltip: Refunds and order editing have been disabled as a result of a lost dispute.

`Won`
No need for notice if we are allowing refunds and editing again.

Jinksi commented on client/order/index.js:
```diff
@@ -129,102 +134,172 @@ jQuery( function ( $ ) {
+	let refundDisabledNotice = '';
+	if ( shouldDisableRefund ) {
+		const refundButton = document.querySelector( 'button.refund-items' );
+		if ( refundButton ) {
```
The refund button isn't rendered for `lost` disputes, so the expected message is not rendered due to this `if` statement.

Jinksi reviewed (changes requested):
I'll change my review status to `Request changes` due to the message missing for `lost` disputes: https://github.com/Automattic/woocommerce-payments/pull/7043#discussion_r1310979141

shendy-a8c committed:
Merge branch 'develop' into update/6378-disable-refund-button-when-disputed

# Conflicts:
#	client/order/index.js

shendy-a8c committed:
Merge branch 'develop' into update/6378-disable-refund-button-when-disputed

shendy-a8c committed:
Import `InlineNotice` from path without `wcpay`.

shendy-a8c committed:
Copy change when refund button is disabled and dispute is under review and lost.

shendy-a8c commented:
Done:
- Copy change per suggested https://github.com/Automattic/woocommerce-payments/pull/7043#issuecomment-1698585823
- Resolve conflict due to `BannerNotice` renamed to `InlineNotice`.

shendy-a8c commented:
@Jinksi I see what happened when dispute is `lost`. In my local, I saw the notice because the Refund button was present.

<img width="620" alt="Screenshot 2023-08-31 at 22 54 17" src="https://github.com/Automattic/woocommerce-payments/assets/73803630/6769cb8f-9e0a-4598-945b-ae18d4df53b8">

On the [demo store](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/post.php?post=106&action=edit), somehow the behaviour is different compared to my local but I think the demo store behaves correctly because as [it receives `charge.dispute.closed` webhook event](https://github.com/Automattic/woocommerce-payments/blob/1ba4330d04f488bcc63a8b691c044a23c783198f/includes/class-wc-payments-webhook-processing-service.php#L159-L161), [it creates a refund and changes the order's status to `Refunded`](https://github.com/Automattic/woocommerce-payments/blob/1ba4330d04f488bcc63a8b691c044a23c783198f/includes/class-wc-payments-webhook-processing-service.php#L587C36-L587C36), so the Refund button does not get rendered.

My local store does not do all that. I think there is a problem with the webhook event handler.

Anyway, in the case where refund button is not there, I think it makes sense to *not* display any notice because we are not disabling the refund button (because it's not there). In terms of notice to give some explanation to what happened, it's available in the order notes.

<img width="295" alt="Screenshot 2023-08-31 at 23 23 15" src="https://github.com/Automattic/woocommerce-payments/assets/73803630/6b44e569-88fe-4646-b33d-555eec177bdf">

What do you think?

shendy-a8c commented:
By the way, I notice some inconsistencies on how to specify the import paths.

https://github.com/Automattic/woocommerce-payments/pull/7078/files#diff-0a103b2ca64b81f1b5b46d347dd571dbcf276eac6ea7fc06222584bb2ba370cdL15 removes the `wcpay/` part, making the import path not absolute.

Jinksi commented on client/order/index.js:
```diff
@@ -129,102 +134,188 @@ jQuery( function ( $ ) {
+			// TODO: finalize text copy after we come up with the final decision.
+			refundDisabledNotice = createInterpolateElement(
+				__(
+					'Refunds and order editing are disabled because this order was disputed. <a>View dispute</a>',
```
It's not clear what dispute status this text will render for, since there is an `if` statement for `lost` disputes further down.

Jinksi committed:
Merge branch 'develop' into update/6378-disable-refund-button-when-disputed

Jinksi reviewed (approved):
🏆 LGTM @shendy-a8c!

I suggest removing unused text (as far as I can tell) in https://github.com/Automattic/woocommerce-payments/pull/7043#discussion_r1314385513, but aside from that, this is looks good to me.

I've checked out this branch on the [Helix demo site](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/admin.php?page=wc-admin&path=/payments/overview) – here are links to disputed orders I've tested:

* [Needs Response](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/post.php?post=137&action=edit)
	* [x] Refunds disabled
	* [x] Notice rendered
	`Refunds and order editing are disabled during disputes.`
* [Under Review](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/post.php?post=108&action=edit)
	* [x] Refunds disabled
	* [x] Notice rendered 
	`This order has an active payment dispute. Refunds and order editing are disabled during this time. View details`
* [Won](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/post.php?post=76&action=edit)
	* [x] No notice
	* [x] Refunds allowed
* [Lost](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/post.php?post=106&action=edit) (losing evidence)
	* [x] Refunds disabled
	* [x] No notice rendered **due to `Refunded` order status**
* [Lost](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/post.php?post=121&action=edit) (accepted)
	* [x] Refunds disabled
	* [x] No notice rendered **due to `Refunded` order status**
* [Inquiry: Needs Response](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/post.php?post=138&action=edit)
	* [x] Refunds allowed
	* [x] Notice rendered
* [Inquiry: Under Review](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/post.php?post=139&action=edit)
	* [x] Refunds allowed
	* [x] No notice rendered
* [Inquiry: Closed](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/post.php?post=74&action=edit) (refunded)
	* [x] Refunds allowed (although $0 left to refund)
	* [x] No notice
* [Inquiry: Closed](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/post.php?post=140&action=edit) (winning evidence)
	* [x] Refunds allowed (with full amount available to refund)
	* [x] No notice

shendy-a8c committed:
Merge branch 'develop' into update/6378-disable-refund-button-when-disputed

shendy-a8c committed:
Remove TODO and remove unused default value for refund disabled notice and tooltip text.

shendy-a8c commented:
Done:
- Removed TODO.
- Removed unused default value for the disabled refund notice and tooltip text.

Also, since we align with the logic where we allow refunds for won dispute (p1693888693029249/1692768707.304979-slack-C03LA109BKM), I think this PR is good to be merged.
